src/components/confirm/confirm.test.tsx

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Confirm from "../confirm";
+
+jest.mock("../../__internal__/utils/helpers/guid", () => () => "guid-12345");
+
+describe("Confirm", () => {
+  it("should render the component with expected title and subtitle", () => {
+    render(
+      <Confirm open title="Title" subtitle="Subtitle" onConfirm={() => {}} />
+    );
+
+    expect(screen.getByText("Title")).toBeVisible();
+    expect(screen.getByText("Subtitle")).toBeVisible();
+  });
+
+  it("should render the confirm button if onConfirm is set", () => {
+    render(<Confirm open onConfirm={() => {}} />);
+
+    expect(screen.getByRole("button", { name: "Yes" })).toBeVisible();
+  });
+
+  it("should render the cancel button if onCancel is set", () => {
+    render(<Confirm open onConfirm={() => {}} onCancel={() => {}} />);
+
+    expect(screen.getByRole("button", { name: "No" })).toBeVisible();
+  });
+
+  it("should call onConfirm when confirm button is clicked", async () => {
+    const onConfirm = jest.fn();
+    const user = userEvent.setup();
+    render(<Confirm open onConfirm={onConfirm} />);
+
+    await user.click(screen.getByRole("button", { name: "Yes" }));
+    expect(onConfirm).toHaveBeenCalledTimes(1);
+  });
+
+  it("should call onCancel when cancel button is clicked", async () => {
+    const onCancel = jest.fn();
+    const user = userEvent.setup();
+    render(<Confirm open onConfirm={() => {}} onCancel={onCancel} />);
+
+    await user.click(screen.getByRole("button", { name: "No" }));
+    expect(onCancel).toHaveBeenCalledTimes(1);
+  });
```

Comment:
suggestion(non-blocking): these four tests should be able to be condensed into two tests

Believe that should be possible anyway, if we lose coverage then obviously not. 

We could also add a more descriptive test title for them as well. e.g.

`it("should render a confirm button when onConfirm prop is set and will execute the function when clicked")`